### PR TITLE
PHOENIX-5960 : Creating view on non-existent table should throw TNFE

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  12m 16s |  master passed  |
| +1 :green_heart: |  compile  |   0m 58s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 57s |  phoenix-core in master has 966 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  phoenix-core: The patch generated 0 new + 159 unchanged - 1 fixed = 159 total (was 160)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  8s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 107m 11s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   | 141m 10s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.ViewMetadataIT |
|   | phoenix.end2end.TableSnapshotReadsMapReduceIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/987 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 0a46bf80e71b 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / cfbd997 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/1/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/1/testReport/ |
| Max. process+thread count | 7047 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 44s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  15m 21s |  master passed  |
| +1 :green_heart: |  compile  |   1m 17s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 44s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m  1s |  master passed  |
| +0 :ok: |  spotbugs  |   3m 45s |  phoenix-core in master has 966 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 52s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 35s |  phoenix-core: The patch generated 2 new + 155 unchanged - 5 fixed = 157 total (was 160)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  3s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 106m 56s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 30s |  The patch does not generate ASF License warnings.  |
|  |   | 146m 53s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.ViewMetadataIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/987 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 656f28297fd8 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / d1fc3f5 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/3/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/3/testReport/ |
| Max. process+thread count | 7037 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by swaroopak — APPROVED** [review]
LGTM, Thanks for the patch @virajjasani

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java [hunk 1]
```diff
@@ -801,6 +802,37 @@ public void testSetTableDescriptorPropertyOnView() throws Exception {
...
+    @Test
+    public void testCreateViewFromNonExistentTable() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute(
+                "CREATE TABLE IF NOT EXISTS S.T1 (A INTEGER PRIMARY KEY, B INTEGER)");
+            // 1. create view from non-existent table (without schema)
+            try {
+                conn.createStatement().execute(
+                    "CREATE VIEW IF NOT EXISTS V1(C INTEGER) AS SELECT * FROM T2");
+            } catch (TableNotFoundException e) {
```
Comment (yanxinyi) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java:814]: can you add a flag that makes sure the code goes to the catch block?
something likes:

flag = false;

try {

} catch {
 flag = true;
}

assertTrue(flag)

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java [hunk 2]
```diff
@@ -801,6 +802,37 @@ public void testSetTableDescriptorPropertyOnView() throws Exception {
...
+                assertEquals("", e.getSchemaName());
+                assertEquals("ERROR 1012 (42M03): Table undefined. tableName=T2",
+                    e.getMessage());
+            }
+            // 2. create view from existing table - successful
+            conn.createStatement().execute(
+                "CREATE VIEW IF NOT EXISTS V1(C INTEGER) AS SELECT * FROM S.T1");
+            // 3. create view from non-existent table (with schema)
+            try {
+                conn.createStatement().execute(
```
Comment (yanxinyi) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java:825]: same as above

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java (hunk 1 above)
Comment (virajjasani) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java:815]: Done, added `fail()`.
Thanks

**Comment by virajjasani** [comment]
Thanks for the reviews @swaroopak @yanxinyi

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java (hunk 2 above)
Comment (virajjasani) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java:826]: Done

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java (hunk 1 above)
Comment (yanxinyi) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java:815]: oh yeah, this is nicer than a flag :)

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 18s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 55s |  master passed  |
| +1 :green_heart: |  compile  |   1m  3s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 47s |  master passed  |
| +0 :ok: |  spotbugs  |   3m 11s |  phoenix-core in master has 966 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  phoenix-core: The patch generated 0 new + 156 unchanged - 4 fixed = 156 total (was 160)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 22s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 210m 49s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m  5s |  The patch does not generate ASF License warnings.  |
|  |   | 251m 13s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/987 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 8e883417f7b0 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / d1fc3f5 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/4/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/4/testReport/ |
| Max. process+thread count | 6113 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  5s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  12m 25s |  master passed  |
| +1 :green_heart: |  compile  |   0m 57s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 33s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 45s |  master passed  |
| +0 :ok: |  spotbugs  |   2m 51s |  phoenix-core in master has 966 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  phoenix-core: The patch generated 0 new + 156 unchanged - 4 fixed = 156 total (was 160)  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  5s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 181m 32s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 27s |  The patch does not generate ASF License warnings.  |
|  |   | 216m 14s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.UpsertSelectIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/5/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/987 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux eab3d6927f6d 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | master / 457a67c |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/5/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/5/testReport/ |
| Max. process+thread count | 6764 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-987/5/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**